marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/IOEndpointImpl.java

```diff
@@ -119,21 +123,23 @@ public void checkAllowedArgs(InputStream endpointState, SessionState session, In
         private byte[]         endpointState;
         private byte[]         workUnit;
         private SessionState   session;
+        private CallContextImpl callContext;
 
         private long           callCount = 0;
 
         BulkIOEndpointCallerImpl(IOEndpointImpl endpoint) {
             if (endpoint == null)
                 throw new IllegalArgumentException("null endpoint definition");
+            this.callContext = new CallContextImpl(endpoint);
             this.endpoint = endpoint;
         }
 
         private IOEndpointImpl getEndpoint() {
-            return this.endpoint;
+            return callContext.getEndpoint();
         }
 
         String getEndpointPath() {
-            return getEndpoint().getEndpointPath();
+            return callContext.getEndpoint().getEndpointPath();
```

Comment:
Would it be better to use the getEndpoint() getter and thus maintain the access to the endpoint in one place?